presto-main/src/main/java/com/facebook/presto/operator/TableScanOperator.java

```diff
@@ -205,6 +209,9 @@ public boolean isFinished()
         if (!finished) {
             createSourceIfNecessary();
             finished = (source != null) && source.isFinished();
+            if (source != null) {
+                systemMemoryContext.setBytes(source.getSystemMemoryUsage());
```

Comment:
Why do we need to update the memory usage here? It seems like the update in `finish()` will take care of it